paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -149,48 +150,99 @@ def clamp_value(number):
     return int(round(clamp_value(Kp * error + iterm + Kd * (error - last_error) / time_delta)))
 
 
-@register_autoscaling_component('http', SERVICE_METRICS_PROVIDER_KEY)
-def http_metrics_provider(marathon_service_config, marathon_tasks, mesos_tasks, endpoint='status', *args, **kwargs):
-    """
-    Gets the average utilization of a service across all of its tasks, where the utilization of
-    a task is read from a HTTP endpoint on the host.
+def get_json_body_from_service(host, port, endpoint):
+    return requests.get(
+        'http://%s:%s/%s' % (host, port, endpoint),
+        headers={'User-Agent': get_user_agent()},
+    ).json()
 
-    The HTTP endpoint must return JSON with a 'utilization' key with a value from 0 to 1.
+
+def get_http_utilization_for_all_tasks(marathon_service_config, marathon_tasks, endpoint, json_mapper):
+    """
+    Gets the mean utilization of a serivce across all of its tasks by fetching
+    json from an http endpoint and applying a function that maps it to a
+    utilization
 
     :param marathon_service_config: the MarathonServiceConfig to get data from
     :param marathon_tasks: Marathon tasks to get data from
-    :param mesos_tasks: Mesos tasks to get data from
+    :param endpoint: The http endpoint to get the uwsgi stats from
+    :param json_mapper: A function that takes a dictionary for a task and returns that task's utilization
 
-    :returns: the service's average utilization, from 0 to 1
+    :returns: the service's mean utilization, from 0 to 1
     """
 
     endpoint = endpoint.lstrip('/')
     utilization = []
     for task in marathon_tasks:
         try:
-            endpoint_utilization = requests.get(
-                'http://%s:%s/%s' % (task.host, task.ports[0], endpoint),
-                headers={'User-Agent': get_user_agent()}
-            ).json()['utilization']
-            utilization.append(float(endpoint_utilization))
-        except Exception:
-            pass
+            utilization.append(json_mapper(get_json_body_from_service(task.host, task.ports[0], endpoint)))
+        except requests.exceptions.Timeout:
+            # If we time out querying an endpoint, assume the task is fully loaded
+            # This won't trigger in the event of DNS error or when a request is refused
+            # a requests.exception.ConnectionError is raised in those cases
+            utilization.append(1.0)
```

Comment:
Can you log.debug on this one as well? 
This is another example of "why is this thing not doing what I think it would do", and logging helps.